src/data/firebase.ts

```diff
@@ -0,0 +1,106 @@
+import { getImportedKey } from "./index";
+import * as firebase from "firebase/app";
+import "firebase/firestore";
+import { ExcalidrawElement } from "../element/types";
+import { getDrawingVersion } from "../element";
+
+const firebaseConfig = JSON.parse(process.env.REACT_APP_FIREBASE_CONFIG);
+
+firebase.initializeApp(firebaseConfig);
+
+interface FirebaseStoredDrawing {
+  drawingVersion: number;
+  ciphertext: firebase.firestore.Blob;
+}
+
+const EMPTY_IV = new Uint8Array(12);
+
+async function encryptElements(
+  key: string,
+  elements: readonly ExcalidrawElement[],
+): Promise<ArrayBuffer> {
+  const importedKey = await getImportedKey(key, "encrypt");
+  const iv = EMPTY_IV;
```

Comment:
The `iv` must never be reused. This is fine for the normal links because we generate a new key encryption each time but not okay for this since we reuse the same room key every time. You need to generate a new one each time.

The `iv` is not a secret, you can put it at the beginning of the payload if that's the easiest way to pass it around.

Reply:
fixed this